cookbooks/lower_crustal_flow/lower_crustal_flow_obstacle.prm

```diff
@@ -0,0 +1,118 @@
+# A simple setup for a lower crustal flow around an obstacle in 3D based on Clark et al, GJI, 2005.
+
+############### Global parameters
+
+set Dimension                              = 3
+set Start time                             = 0
+set End time                               = 0
+set Use years in output instead of seconds = true
+set Output directory                       = output-lower-crustal-flow-obstacle2
+set Pressure normalization                 = volume
+set Timing output frequency                = 1
+
+############### Parameters of the linear solver
+
+subsection Solver parameters
+  subsection Stokes solver parameters
+    set Number of cheap Stokes solver steps = 0
+    set Maximum number of expensive Stokes solver steps = 3000
+  end
+  subsection Stokes solver parameters
+    set Stokes solver type = block GMG
+  end
+end
+
+#subsection Discretization
+#  set Stokes velocity polynomial degree       = 2
+#  set Use locally conservative discretization = true
+#end
+
+
+############### Parameters describing the model
+
+subsection Geometry model
+  set Model name = box
+  subsection Box
+    set X extent  = 1200e3
+    set Y extent  = 600e3
+    set Z extent  = 15e3
+    set X repetitions  = 10
+    set Y repetitions  = 5
+  end
+end
+
+subsection Compositional fields
+  set Number of fields = 1
+end
+
+# first  material is lower crust
+# second material is obstacle
+
+subsection Initial composition model
+  set Model name = function
+  subsection Function
+    set Variable names      = x,y,z
+    set Function constants  = a=200e3
+    set Function expression = if ((x-600e3)^2+(y)^2 < a^2,1,0)
+  end
+end
+
+subsection Material model
+  set Model name = simple
+  subsection Simple model
+    set Reference density = 3000
+    set Viscosity = 2e18
+    set Density differential for compositional field 1 = 10
```

Comment:
Just to avoid confusion, I would remove this line (default is zero) since the density is not important anyway.